runtime/src/commonJvmAndroid/kotlin/pbandk/MessageExtensionsJvm.kt

```diff
@@ -34,3 +29,70 @@ public fun <T : Message> Message.Companion<T>.decodeFromStream(stream: InputStre
  */
 public fun <T : Message> Message.Companion<T>.decodeFromByteBuffer(buffer: ByteBuffer): T =
     decodeWith(BinaryMessageDecoder.fromByteBuffer(buffer))
+
+/**
+ * Read the next message of type T, in the [stream] of size-delimited Messages, such as is written by the Java method
+ * AbstractMessageLite.writeDelimitedTo(OutputStream) and read by Message.parseDelimitedFrom(InputStream).
+ * A null is returned when all messages have been read, and the stream is closed.
+ */
+public fun <T : Message> Message.Companion<T>.parseDelimitedFrom(stream: InputStream): T? {
+    val length = readVlen(stream)
+    if (length < 0) {
+        stream.close()
+        return null
+    }
+    val message = stream.readNBytes(length)
+    if (message.size != length) {
+        throw IOException("damaged delimited stream")
+    }
+    return this.decodeFromByteArray(message)
+}
+
+// variable length (base 128) integer returned as an Int
+private fun readVlen(input: InputStream): Int {
```

Comment:
Ive sent you a possible solution. It seemed pretty elegant until I realized readerRawVarint64 needed an eof indication not encoded in the byte return value.  One could also follow java InputStream and use input.read() to return an unsigned Byte in an Int and use -1 as the eof indicator. 

I thought Id send you a possible solution and get your input before going down any rabbit holes.

Reply:
Also youll see that I didnt modify KotlinBinaryWireEncoder but did modify KotlinBinaryWireDecoder to make sure the closure worked with existing code. 

Havent run any extensive tests, waiting your input.

Reply:
Your overall solution looks good 👍🏼 I left some comments about small bugs/issues that need to be addressed.

Also, rather than having `readByte()` return a `ReturnRawByte` structure, I think you could just have the function `throw InvalidProtocolBufferException.truncatedMessage()` on EOF (see my other comment about this). Then `readByte()` can just the `Byte` directly without wrapping it in a data class.